### PR TITLE
drainer: support load table infos to save memory and add gc safepoint update

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb-binlog/issues/1137

### What is changed and how it works?
1. support load table infos from tidb
2. add gc safepoint ttl update to avoid table info is gced

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### Release note

- No release note

### CODE REVIEW

File: drainer/safepoint.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+)
+
+const (
+	drainerServiceSafePointPrefix = "drainer"
+	defaultDrainerGCSafePointTTL  = 5 * 60
+)
+
+func updateServiceSafePoint(ctx context.Context, pdClient pd.Client, cpt checkpoint.CheckPoint, ttl int64) {
+	updateInterval := time.Duration(ttl/2) * time.Second
+	tick := time.NewTicker(updateInterval)
```
Comment (lance6716) [inline drainer/safepoint.go:21]: tick.Stop

**Review by lance6716 — COMMENTED** [review]
rest lgtm

File: drainer/safepoint.go (hunk 1 above)
Comment (lichunzhu) [reply drainer/safepoint.go:21]: addressed in https://github.com/pingcap/tidb-binlog/pull/1233/commits/fcd27dc32973a449dd6b94e0bb136c7a7fd6d9a3

**Review by lance6716 — APPROVED** [review]

**Pull request merged**